js/src/framework.ts

```diff
@@ -140,9 +161,14 @@ declare global {
 
 globalThis._evals = {};
 
-export async function Eval<Input, Output, Expected>(
+export async function Eval<
+  Input,
+  Output,
+  Expected,
+  Metadata extends BaseMetadata,
```

Comment:
I think this breaks existing evals which don't specify anything for metadata. When I build this branch and compile one of our internal tests, I see the following error:
```
tests/expect_tests/basic.eval.ts:5:3 - error TS2322: Type '() => Promise<{ input: string; expected: string; }[]>' is not assignable to type 'EvalData<string, string, object>'.
  Type '() => Promise<{ input: string; expected: string; }[]>' is not assignable to type '() => EvalCase<string, string, object>[]'.
    Type 'Promise<{ input: string; expected: string; }[]>' is missing the following properties from type 'EvalCase<string, string, object>[]': length, pop, push, concat, and 29 more.

5   data: async () => {
    ~~~~

  sdk/js/dist/framework.d.ts:37:5
    37     data: EvalData<Input, Expected, Metadata>;
           ~~~~
    The expected type comes from property 'data' which is declared here on type 'Evaluator<string, string, string, object>'
```

The chunk it's complaining about is here:
```
  Eval("basic", {
E   data: async () => {     ■ Type '() => Promise<{ input: string; expected: string; }[]>' is not assignable to type 'EvalData<string, string, object>'.    Type '() => Promise<{ input: string; expected: string; }[]>' is not assignable to type '() => EvalCase<string, string, object>[]'.      Type 'Promise<{ input: string; expected: string; }[]>' is missing the
      return [
        {
          input: "c",
          expected: "d",
        },
      ];
    },
```

My guess is that restricting `Metadata` to be a subtype of `BaseMetadata` prevents it from being nothing (`void`). It seems this can be fixed by changing `BaseMetadata = object` to `BaseMetadata = object | void`.

Reply:
Also I think this will have to be defaulted to `Metadata extends BaseMetadata = void`, otherwise existing callsites which have type specs for eval as `Eval<something, something, something>` will complain that there is no fourth argument.

Reply:
Updated the type signature and updated  `BaseExperiment` and `Eval` to set a default value of void. Do you think we need to update other places too?

Reply:
In theory, to avoid any breakage, we should include defaults for every publicly-exported type/function signature which adds a `Metadata` param. But if we don't expect much breakage outside of `BaseExperiment` and `Eval`, maybe it's not worth the clutter.

Reply:
ok will update!